include/target.mk

```diff
@@ -68,6 +68,9 @@ define Profile
   $(eval $(call ProfileDefault))
   $(eval $(call Profile/$(1)))
   dumpinfo : $(call shexport,Profile/$(1)/Description)
+  _UNWANTED := $(filter -%,$(PACKAGES))
+  UNWANTED := $(patsubst -%,%,$(_UNWANTED))
+  PACKAGES := $(filter-out $(UNWANTED),$(DEFAULT_PACKAGES)) $(filter-out $(_UNWANTED),$(PACKAGES))
```

Comment:
there is already a `filter_packages` helper a few lines above that may do exactly what you are doing here [ i hope i did not misread ; makefile magic is not my forte ]

also, IIRC, specifying `DEFAULT_PACKAGES += -pkg` should already remove a package ;
this is target-wide of course ;
i used this recently and it worked ;

does it already work to just add a `-pkg` somewhere in your profile ?
like 
```
define Profile/xxx
   PACKAGES+= -pkg
endef
```
i could be wrong here though

Reply:
> does it already work to just add a -pkg somewhere in your profile ?

Yes it work this way and [is used already](https://github.com/openwrt/openwrt/blob/master/target/linux/lantiq/image/Makefile#L156). I unpacked the images and it doesn't have `/sbin/swconfig`

Reply:
That is DEVICE_PACKAGES though, which is separate from the profile PACKAGES as far as I realize?

I don't see how this patch would work though - it replaces PACKAGES with a list of packages generated from DEFAULT_PACKAGES and PACKAGES, while before, PACKAGES was simply set to a value in profiles, without taking DEFAULT_PACKAGES into consideration at all.

Reply:
@commodo Except that `filter_packages` doesn't seem to be used anywhere.

@NeoRaider Updated to not combine `DEFAULT_PACKAGES` and `PACKAGES` back into `PACKAGES`, but to keep them separate.

Reply:
@pprindeville I'm quite sure I don't get your usecase or constrains. De/unselecting packages, selected via DEFAULT_PACKAGES, works as described by comodo.